resources/objects/furniture/food_bucket.gd

```diff
@@ -4,9 +4,17 @@ extends Node3D
 
 
 func _on_exit_detector_body_exited(body):
-	body = body as Food
-	if(body and !body.has_left_spawner):
-		body.has_left_spawner = true
-		var item = item_scene.instantiate()
-		add_child(item)
-		item.global_position = body.position_before_pickup
+	if body is Food:
+		body = body as Food
+		if(body and !body.has_left_spawner):
+			body.has_left_spawner = true
+			var item = item_scene.instantiate()
+			add_child(item)
+			item.global_position = body.position_before_pickup
+	else:
+		body = body as XRToolsPickable
+		if(body and !body.has_left_spawner):
+			body.has_left_spawner = true
+			var item = item_scene.instantiate()
+			add_child(item)
+			item.global_position = body.position_before_pickup
```

Comment:
Don't duplicate code. Just delete the line `body = body as Food`.

Reply:
Fair